packages/replayor/benchmark.go

```diff
@@ -301,18 +314,23 @@ func (r *Benchmark) Run(ctx context.Context) {
 	for {
 		select {
 		case <-doneChan:
-			r.log.Info("writing block stats")
 			r.s.Write(ctx)
 			return
 		case <-ticker.C:
-			currentBlock, err := r.clients.DestNode.BlockByNumber(ctx, nil)
+			currentBlock, err := r.getLatestBlockFromDestNode(ctx)
```

Comment:
The underlying method here uses `retry.Do` to give some retries. Makes the test more robust and less likely to exit prematurely